### PR TITLE
Create DFK seed liquidity actions table

### LINKED ISSUE #25: Table: DFK_Seed_Liquidity_Added
Let's create a `DFK_Seed_Liquidity_Added` table to trace all DFK Added Liquidity Transactions.
This can be created using `txs+logs` as the source data table.

Hints:
```
+ " WITH LP_INTERESTED as ("
+ " select pool_address, pool_name, token0, token1"
+ " from harmony.dev.ant_dex_lp_labels"
+ " )"
+ " , tx_interested as ("
+ " select tx_id from harmony.dev.ant_view_txs"
+ " where 1=1"
+ " and eth_to_address = '0x24ad62502d1c652cc7684081169d04896ac20f30'"
+ " and eth_from_address = :1"
+ " and substr(input,0,10) in ('0x02751cec','0xf305d719','0xbaa2abde','0xe8e33700')"
+ " )"
+ " , Add_LP as ("
+ " select"
+ " ma.block_timestamp as block_timestamp"
+ " ,ma.tx_id as tx_id"
+ " ,ma.event_inputs:amount0 / pow(10, t0.decimals) as amount_0"
+ " ,ma.event_inputs:amount1 / pow(10, t1.decimals) as amount_1"
+ " ,ma.eth_contract_address as lp_address"
+ " ,dex.token0 as t0_address"
+ " ,dex.token1 as t1_address"
+ " ,t0.token_name as t0_name"
+ " ,t1.token_name as t1_name"
+ " ,CASE"
+ " WHEN t0_address = '0x72cb10c6bfa5624dd07ef608027e366bd690048f' THEN amount_0 * 2 * jewel.price"
+ " WHEN t1_address = '0x72cb10c6bfa5624dd07ef608027e366bd690048f' THEN amount_1 * 2 * jewel.price"
+ " WHEN t0_address = '0xcf664087a5bb0237a0bad6742852ec6c8d69a27a' THEN amount_0 * 2 * wone.price"
+ " WHEN t1_address = '0xcf664087a5bb0237a0bad6742852ec6c8d69a27a' THEN amount_1 * 2 * wone.price"
+ " END as amount_usd"
+ " from harmony.dev.ant_logs ma"
+ " LEFT JOIN harmony.dev.ant_dex_lp_labels dex on ma.eth_contract_address = dex.pool_address"
+ " LEFT JOIN harmony.dev.ant_tokens2 t0 on dex.token0 = t0.token_address"
+ " LEFT JOIN harmony.dev.ant_tokens2 t1 on dex.token1 = t1.token_address"
+ " LEFT JOIN harmony.dev.ant_view_token_price_jewel jewel on date(jewel.timestamp) = date(ma.block_timestamp)"
+ " LEFT JOIN harmony.dev.ant_view_token_price_one wone on date(wone.timestamp) = date(ma.block_timestamp)"
+ " where 1=1"
+ " and tx_id in (select tx_id from tx_interested)"
+ " and (event_emitted_name = 'Mint')"
+ " )"
+ " , Remove_LP as ("
+ " select"
+ " ma.block_timestamp as block_timestamp"
+ " ,ma.tx_id as tx_id"
+ " ,ma.event_inputs:amount0 / pow(10, t0.decimals) as amount_0"
+ " ,ma.event_inputs:amount1 / pow(10, t1.decimals) as amount_1"
+ " ,ma.eth_contract_address as lp_address"
+ " ,dex.token0 as t0_address"
+ " ,dex.token1 as t1_address"
+ " ,t0.token_name as t0_name"
+ " ,t1.token_name as t1_name"
+ " ,CASE "
+ " WHEN t0_address = '0x72cb10c6bfa5624dd07ef608027e366bd690048f' THEN amount_0 * 2 * jewel.price"
+ " WHEN t1_address = '0x72cb10c6bfa5624dd07ef608027e366bd690048f' THEN amount_1 * 2 * jewel.price"
+ " WHEN t0_address = '0xcf664087a5bb0237a0bad6742852ec6c8d69a27a' THEN amount_0 * 2 * wone.price"
+ " WHEN t1_address = '0xcf664087a5bb0237a0bad6742852ec6c8d69a27a' THEN amount_1 * 2 * wone.price"
+ " END as amount_usd"
+ " from harmony.dev.ant_logs ma"
+ " LEFT JOIN harmony.dev.ant_dex_lp_labels dex on ma.eth_contract_address = dex.pool_address"
+ " LEFT JOIN harmony.dev.ant_tokens2 t0 on dex.token0 = t0.token_address"
+ " LEFT JOIN harmony.dev.ant_tokens2 t1 on dex.token1 = t1.token_address"
+ " LEFT JOIN harmony.dev.ant_view_token_price_jewel jewel on date(jewel.timestamp) = date(ma.block_timestamp)"
+ " LEFT JOIN harmony.dev.ant_view_token_price_one wone on date(wone.timestamp) = date(ma.block_timestamp)"
+ " where 1=1"
+ " and tx_id in (select tx_id from tx_interested)"
+ " and (event_emitted_name = 'Burn')"
+ " )"
+ " select * from Add_LP"
+ " order by block_timestamp asc"
```

**To Do Next:**

- [ ] Design the schema for this table
- [ ] Get confirmation on schema
- [ ] Build schema
- [ ] Schema build review
- [ ] Submit Pull Request

### PR DESCRIPTION
**Pull request by D0R11**
# Description

Builds the Seed Liquidity Actions table for DFK.

# Test

Here's the sample output running in locally:

```16:45:01 | Concurrency: 4 threads (target='dev')
16:45:01 | 
16:45:01 | 1 of 1 START incremental model DEV.dfk_seed_liquidity_actions........ [RUN]
16:45:11 | 1 of 1 OK created incremental model DEV.dfk_seed_liquidity_actions... [SUCCESS 1 in 9.48s]
16:45:11 | 
16:45:11 | Finished running 1 incremental model in 15.48s.

Completed successfully

Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1```

### CODE REVIEW

**Comment by antonyip** [comment]
Hello @D0R11, please follow the style guide located here: https://github.com/dbt-labs/corp/blob/main/dbt_style_guide.md

**Comment by D0R11** [comment]
Apologies @antonyip, applied the changes now. Kindly check if you have the time. Thank you.

File: models/dfk/dfk_seed_liquidity_actions.sql [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        end as action
+
+    from logs
+        left join liquidity_pools 
+            on logs.evm_contract_address = liquidity_pools.pool_address
+        left join tokens t0 
+            on liquidity_pools.token0 = t0.token_address
+        left join tokens t1 
+            on liquidity_pools.token1 = t1.token_address
+    where 1=1
```
Comment (antonyip) [inline models/dfk/dfk_seed_liquidity_actions.sql]: remove 1=1

File: models/dfk/dfk_seed_liquidity_actions.sql [hunk 2]
```diff
@@ -0,0 +1,82 @@
+{{
+    config(
+        materialized='incremental',
+        unique_key='log_id',
+        tags=['dfk', 'dfk_seed_liquidity_actions'],
+        cluster_by=['log_id']
```
Comment (antonyip) [inline models/dfk/dfk_seed_liquidity_actions.sql]: cluster by block_timestamp

File: models/dfk/dfk_seed_liquidity_actions.sql [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+),
+
+txs as (
+    select 
+        tx_hash 
+    from {{ ref('txs') }}
+    where {{ incremental_load_filter("block_timestamp") }}
+        and to_address = '0x24ad62502d1c652cc7684081169d04896ac20f30'
+        and substr(data,0,10) 
+            in ('0x02751cec','0xf305d719','0xbaa2abde','0xe8e33700')
```
Comment (antonyip) [inline models/dfk/dfk_seed_liquidity_actions.sql]: leave a comment on what these function signatures are

File: models/dfk/dfk_seed_liquidity_actions.sql [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    select * 
+    from {{ ref('tokens') }}
+),
+
+txs as (
+    select 
+        tx_hash 
+    from {{ ref('txs') }}
+    where {{ incremental_load_filter("block_timestamp") }}
+        and to_address = '0x24ad62502d1c652cc7684081169d04896ac20f30'
```
Comment (antonyip) [inline models/dfk/dfk_seed_liquidity_actions.sql]: leave a comment on what this address is

**Review by antonyip — CHANGES_REQUESTED** [review]
some minor changes and we're good to go

**Comment by D0R11** [comment]
Applied the latest changes

File: models/dfk/dfk_seed_liquidity_actions.sql [hunk 5]
```diff
@@ -73,7 +76,6 @@ final_table as (
             on liquidity_pools.token0 = t0.token_address
         left join tokens t1 
             on liquidity_pools.token1 = t1.token_address
-    where 1=1
         and tx_hash in (select tx_hash from txs)
```
Comment (antonyip) [inline models/dfk/dfk_seed_liquidity_actions.sql]: If you remove the where, it becomes an invalid sql..

Comment (D0R11) [reply models/dfk/dfk_seed_liquidity_actions.sql]: Apologies! Added the where clause back

**Review by antonyip — APPROVED** [review]
Looks good to me.
Follow up Issues: https://github.com/MetricsDAO/harmony_dbt/issues/110

**Pull request merged**